### ISSUE TITLE
[Bug] Display handling / data retrieval for (quick) test results when there is no internet connection ...

### ISSUE THREAD

**Issue by Jo-Achim** [corona-warn-app/cwa-app-android#4338]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Unknown for iOS.
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Nokia 7.2, TA-1196, Build: 00WW_2_410_SP06
- Android version: 10
- App version:  2.13.2
- ENF version:  18213961000

## Describe the bug

Somehow it seems unimaginable for designers, developers, programmers and testers these days that a device has no internet connection ;-).

For example, if a quick test result is stored in the CWA, it will be downloaded again each time the CWA is called up.
If there is no internet connection, this leads to the following results:

"Data retrieval ...":
![Screenshot_20211107-133943](https://user-images.githubusercontent.com/21124877/140646242-faae5a61-5d15-4672-9f93-9abd50bbe520.png)

"Quick test error" (after several minutes):
![Screenshot_20211107-134353](https://user-images.githubusercontent.com/21124877/140646252-aa78e513-4bc9-4a4e-ad95-52394670639e.png)

_Option 1:_
Tab on DETAILS:
![Screenshot_20211107-134404](https://user-images.githubusercontent.com/21124877/140646256-a78d71c3-2cd4-4eb9-b41f-8561da248aff.png)

and after tab on "<", the quick test result is displayed:
![Screenshot_20211107-134417](https://user-images.githubusercontent.com/21124877/140646259-455b9d4a-700a-4e92-8aa2-05d244a46218.png)

_Option 2:_
After tab on OK (insteed of DETAILS) in the "quick test error" display, there is either another "data retrieval ..." (see above) or the quick test result is displayed immediately (see last screen sheet in _Option 1_).

## Steps to reproduce the issue

Make sure that a QR code from a quick test has already been scanned into the CWA.

_Option 1:_
1. Start CWA
2. See main screen ("Data retrieval ..."). Wait for error message.
3. Tap on DETAILS (error message) then tap "<" (back) and
4. the quick test result is displayed (last screenshot above).

_Option 2:_
1. Start CWA again.
2. See main screen ("Data retrieval ..."). Wait for error message.
3. Tap on OK (error message), the "data retrieval ..." may be repeated (wait again) and / or 
4. the quick test result is displayed (last screenshot above).

It seems that after a certain time the "data retrieval ..." starts again automatically, even if the "quick test result" was displayed last.

## Expected behaviour

Shouldn't the certificate be displayed as in the last screenshot if there is no internet connection?
Or a possibly necessary check is attempted in the background?

## Possible Fix

It should not only annoy the gate keeper a bit if he has to wait minutes for the necessary display of a quick test result.
And it is unlikely to be conducive to acceptance of the CWA solution.

## Additional context

Are only rapid test results affected by this - without internet connection - issue?

**Comment by Jo-Achim** [comment]
If there is no internet connection when the CWA is starting, the above also applies to PCR-Tests.

**Comment by larswmh** [comment]
Just to make sure I understand you correctly, when you're using "certificate" in this context, you're not referring to the DCCs but to the "test proof"/Testnachweis? Internal Tracking ID: [EXPOSUREAPP-10746](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10746)

---
Corona-Warn-App Open Source Team

**Comment by Jo-Achim** [comment]
@larswmh,
yes, sorry, I mean the tests (rapid test / PCR test) on main screen; according to the screenshots. Not the DCCs.

**Comment by Jo-Achim** [comment]
@larswmh,
the "rapid-test" label is wrong, because ...
> If there is no internet connection when the CWA is starting, the above also applies to PCR-Tests.

the problem affects 'rapid-tests' and 'PCR-tests'.

**Comment by Ein-Tim** [comment]
@Jo-Achim Can you still reproduce this with version 2.17.1

**Comment by dsarkar** [comment]
@Jo-Achim @Ein-Tim With 2.16 behaviour should be: The "Details" button on the dialog has been removed, you can only tap OK. The test result seems to show up in flight mode right from the beginning. It's not trying to calculate the result anymore (negative rat and pending pcr should show up fine).

**Comment by Jo-Achim** [comment]
@dsarkar,

no, I can't verify the problem at the moment because I don't currently have a quick test / no quick test is open.

**Comment by Jo-Achim** [comment]
_I'd like to add a point that goes with the title here... (and I'm not sure if it's already been addressed)_

If the CWA starts, the result of a (quick) test is called up automatically. This can take a moment - especially if the smartphone is abroad.
So far so good.

If one then pulls up the details of the test result and returns back to the CWA main screen (via X), the same test result that was just retrieved is obviously retrieved again.
I'm not sure if this is really necessary!?

**Comment by dsarkar** [comment]
PR https://github.com/corona-warn-app/cwa-app-android/pull/5223
